configuration/src/main/java/io/camunda/configuration/UnifiedConfigurationHelper.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.configuration;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.core.env.Environment;
+
+@ComponentScan("io.camunda.configuration")
+public class UnifiedConfigurationHelper {
+
+  private static final Map<String, Set<String>> LEGACY_PROPERTIES =
```

Comment:
:wrench: This would become too huge when we add more and more properties? Can you think of a way to modularize them? May be this map could be defined in each config class where the new property is defined. The validate method takes it as one of the parameter.

Reply:
I thought about that approach as well, but I dislike that the properties are spread instead of being in one single place.

Maybe we could move the dictionary to its own class? WDYT